jni/lib/src/array.dart

```diff
@@ -4,17 +4,25 @@
 
 // ignore_for_file: unnecessary_cast
 
-part of 'jni_object.dart';
+part of 'types.dart';
```

Comment:
Some of the files in this folder have a plural name, and some singular.

I don't see any style guide on names: https://dart.dev/guides/language/effective-dart/style#do-name-libraries-and-source-files-using-lowercase_with_underscores

The Java style is a single class per file and have the filename correspond.
I believe we do the same in Dart often if we have a single class in a file.
If we have more than one class in a file, then we name it after the most important class if there is one. If there is no most important one it probably has a name reflecting the group of classes.

jarray.dart
jexception.dart (with a JException class that all other exceptions extend)
jobject.dart
jprimitives.dart
jreference.dart
jstring.dart

(Feel free to do this in a separate PR though. If so, file an issue, and we can do it for the whole project at once.)

Reply:
What about `types.dart`? I can add `j`s to all the names

Reply:
`types.dart` here acts like `dart:core`'s `core.dart`. For now I'll add a `j` prefix and have all to be a part of `types.dart` and if you have some other suggestion, please file an issue.